File: design-probes/software-factory/flows/inbox-producer/flow.ts
#!/usr/bin/env deno

// DESIGN PROBE ONLY: hypothetical SDK, complete pseudocode behavior.
import {
  type JsonValue,
  serve,
} from "@flow/run";

interface InboxInput {
  readonly item: string;
}

serve<InboxInput, Record<string, never>>(async run => {
  const inbox = run.attachment("inbox");
  const request = await Deno.readTextFile(
    inbox.resolve(run.input.item),
  );

  const event = await run.effects.call<JsonValue>({
    operationId: "publish-inbox-item",
    slot: "journal",
    method: "append",
    input: {
      type: "https://jig.example/events/inbox-item-created",
      subject: run.input.item,
      data: {
        item: run.input.item,
        request,
      },
    },
  });

  return {
    outcome: "done",
    output: { event },
  };
});

File: design-probes/cordis-timer/flows/schedule-reminder/flow.py
#!/usr/bin/env python3

# DESIGN PROBE ONLY: hypothetical FLOW Run SDK.
from flow_run import serve


async def schedule_reminder(run):
    record = await run.effects.call(
        operation_id="schedule-reminder",
        slot="scheduler",
        method="schedule",
        input=run.input,
    )
    return {"outcome": "done", "output": record}


serve(schedule_reminder)

